### ISSUE TITLE
Test failures on macOS

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#27540]
Testing #27456 on macOS:

Are all of the following 22 test failures really a result of #27452?

```
Running tests...




  Smoke Test Suite

    Data Migration

      1) checks if the Untitled file is restored migrating from stable to latest

      ✓ checks if the newly created dirty file is restored migrating from stable to latest

      ✓ cheks if opened tabs are restored migrating from stable to latest

    Data Loss

      ✓ verifies that 'hot exit' works for dirty files

      2) verifies that contents of the dirty files are restored after 'hot exit'

    Explorer

      ✓ quick open search produces correct result

      ✓ quick open respects fuzzy matching

    Configuration and views

      ✓ turns off editor line numbers and verifies the live change

      ✓ changes 'workbench.action.toggleSidebarPosition' command key binding and verifies it (27655ms)

    Search

      ✓ searches for body & checks for correct result number

      ✓ searches only for *.js files & checks for correct result number

      ✓ dismisses result & checks for correct result number

      ✓ replaces first search result with a replace term

    CSS

      3) verifies quick outline

      4) verifies warnings for the empty rule

      5) verifies that warning becomes an error once setting changed

    JavaScript

      6) shows correct quick outline

      7) finds 'All References' to 'app'

      8) renames local 'app' variable

      ✓ folds/unfolds the code correctly

      9) verifies that 'Go To Definition' works

      10) verifies that 'Peek Definition' works

    Debugging JavaScript

      11) autodetects program attribute for launch.json

      ✓ can set a breakpoint and verify if it's set

    Git

      12) verifies current changes are picked up by Git viewlet

      13) verifies 'app.js' diff viewer changes

      14) stages 'app.js' changes and checks stage count

      15) stages, commits change to 'app.js' locally and verifies outgoing change

    Integrated Terminal

      ✓ opens terminal, runs 'echo' and verifies the output

    Status Bar

      16) verifies presence of all default status bar elements

      17) verifies that 'quick open' opens when clicking on 'Branch', 'Indentation Status, 'Encoding', 'EOL' and 'Language' status elements

      ✓ verifies that 'Problems View' appears when clicking on 'Problems' status element

      ✓ verifies that 'Tweet us feedback' pop-up appears when clicking on 'Feedback' icon

      ✓ checks if 'Go to Line' works if called from the status bar

      ✓ verifies if changing EOL is reflected in the status bar

    Tasks

      18) verifies that eslint task results in 1 problem

      19) is able to select 'Git' output

      20) ensures that build task produces errors in index.js

      21) verifies build errors are reflected in 'Problems View'

    Localization

      22) starts with 'DE' locale and verifies title and viewlets text is in German



  Extensions

    ✓ installs 'vscode-icons' extension and verifies reload is prompted

    ✓ installs an extension and checks if it works on restart





  20 passing (34m)

  22 failing



  1) Smoke Test Suite Data Migration checks if the Untitled file is restored migrating from stable to latest:
     AssertionError: undefined == true
      at Context.<anonymous> (out/tests/data-migration.js:48:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/data-migration.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  2) Smoke Test Suite Data Loss verifies that contents of the dirty files are restored after 'hot exit':

      AssertionError: 'Hello, Code' == 'Hello, Code!'
      + expected - actual

      -Hello, Code
      +Hello, Code!
      
      at Context.<anonymous> (out/tests/data-loss.js:76:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/data-loss.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  3) Smoke Test Suite CSS verifies quick outline:

      AssertionError: 1 == 2
      + expected - actual

      -1
      +2
      
      at Context.<anonymous> (out/tests/css.js:43:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/css.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)

  4) Smoke Test Suite CSS verifies warnings for the empty rule:
     AssertionError: undefined == true
      at Context.<anonymous> (out/tests/css.js:52:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/css.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  5) Smoke Test Suite CSS verifies that warning becomes an error once setting changed:
     AssertionError: undefined == true
      at Context.<anonymous> (out/tests/css.js:65:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/css.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  6) Smoke Test Suite JavaScript shows correct quick outline:

      AssertionError: 1 == 12
      + expected - actual

      -1
      +12
      
      at Context.<anonymous> (out/tests/javascript.js:43:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/javascript.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  7) Smoke Test Suite JavaScript finds 'All References' to 'app':
     An element could not be located on the page using the given search parameters (".reference-zone-widget.results-loaded .peekview-title .meta").
  Error


  8) Smoke Test Suite JavaScript renames local 'app' variable:

      AssertionError: 'anewApp' == 'newApp'
      + expected - actual

      -anewApp
      +newApp
      
      at Context.<anonymous> (out/tests/javascript.js:64:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/javascript.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)

  9) Smoke Test Suite JavaScript verifies that 'Go To Definition' works:
     AssertionError: undefined == true
      at Context.<anonymous> (out/tests/javascript.js:87:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/javascript.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)


  10) Smoke Test Suite JavaScript verifies that 'Peek Definition' works:
     An element could not be located on the page using the given search parameters (".reference-zone-widget.results-loaded .filename").
  Error

  11) Smoke Test Suite Debugging JavaScript autodetects program attribute for launch.json:
     An element could not be located on the page using the given search parameters (".view-lines>:nth-child(11) .mtk7").
  Error


  12) Smoke Test Suite Git verifies current changes are picked up by Git viewlet:
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)
  


  13) Smoke Test Suite Git verifies 'app.js' diff viewer changes:
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)
  


  14) Smoke Test Suite Git stages 'app.js' changes and checks stage count:
     An element could not be located on the page using the given search parameters.
  Error


  15) Smoke Test Suite Git stages, commits change to 'app.js' locally and verifies outgoing change:
     An element could not be located on the page using the given search parameters.
  Error


  16) Smoke Test Suite Status Bar verifies presence of all default status bar elements:

      AssertionError: false == true
      + expected - actual

      -false
      +true
      
      at Context.<anonymous> (out/tests/statusbar.js:40:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/statusbar.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)

  17) Smoke Test Suite Status Bar verifies that 'quick open' opens when clicking on 'Branch', 'Indentation Status, 'Encoding', 'EOL' and 'Language' status elements:
     An element could not be located on the page using the given search parameters.
  Error


  18) Smoke Test Suite Tasks verifies that eslint task results in 1 problem:
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)
  


  19) Smoke Test Suite Tasks is able to select 'Git' output:
     An element could not be located on the page using the given search parameters.
  Error


  20) Smoke Test Suite Tasks ensures that build task produces errors in index.js:
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)
  

  21) Smoke Test Suite Tasks verifies build errors are reflected in 'Problems View':
     Error: the string "Could not retrieve the element in 45 seconds." was thrown, throw an Error :)
  


  22) Smoke Test Suite Localization starts with 'DE' locale and verifies title and viewlets text is in German:

      AssertionError: 'quellcodeverwaltung' == 'quellcodeverwaltung: git'
      + expected - actual

      -quellcodeverwaltung
      +quellcodeverwaltung: git
      
      at Context.<anonymous> (out/tests/localization.js:42:24)
      at Generator.next (<anonymous>)
      at fulfilled (out/tests/localization.js:8:58)
      at process._tickDomainCallback (internal/process/next_tick.js:129:7)






npm ERR! Test failed.  See above for more details.
```

**Comment by michelkaporin** [comment]
@weinand Yes, this amount is due to extension host that is not starting up. As VS Code UI is dependent on it, it makes very many tests to fail.

I would expect maximum 1-3 failures out of all these in a pessimistic case, once #27452 is fixed.

**Comment by weinand** [comment]
Is there a workaround for #27452?
E.g. I could make the data folder non-empty

**Comment by michelkaporin** [comment]
@weinand answered in #27456 😉

**Comment by michelkaporin** [comment]
Has been fixed via #27452 per https://github.com/Microsoft/vscode/issues/27456#issuecomment-305437505, closing.
